internal/web_server/web/handlers/api/admin_api.go

```diff
@@ -345,12 +366,79 @@ func (aah *AdminApiHandler) createQuestion(c echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, errorText)
 	}
 
-	// Save the question to the database
-	err = questions.AddNewQuestion(aah.sharedData.DB, question)
-	if err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, "Failed to create new question")
+	// Get the question by ID from the database.
+	tempQuestion, err := questions.GetQuestionByID(aah.sharedData.DB, questionID)
+
+	// If doesn't exist in the database.
+	if err != nil && err == sql.ErrNoRows {
+		// Save the question to the database.
+		err = questions.AddNewQuestion(aah.sharedData.DB, question)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to create new question")
+		}
+	} else if tempQuestion.ID == questionID {
+		// If the question ID is found, update the question.
+		question.ID = questionID
+		err = questions.UpdateQuestion(aah.sharedData.DB, &question)
 	}
 
-	// Return the "question item" element
+	// Return the "question item" element.
 	return utils.Render(c, http.StatusOK, dashboard_components.QuestionListItem(&question))
 }
+
+// Delete a question with the given ID from the database.
+func (aah *AdminApiHandler) deleteQuestion(c echo.Context) error {
+	// Get the question ID
+	questionID, err := uuid.Parse(c.QueryParam(queryParamQuestionID))
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, errorInvalidQuestionID)
+	}
+
+	// Delete the question from the database
+	err = questions.DeleteQuestionByID(aah.sharedData.DB, questionID)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Failed to delete question")
+	}
+
+	return c.NoContent(http.StatusOK)
+}
+
+// Edit the image for a question in the database.
+func (aah *AdminApiHandler) editQuestionImage(c echo.Context) error {
+	// Get the question ID
+	questionID, err := uuid.Parse(c.QueryParam(queryParamQuestionID))
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, errorInvalidQuestionID)
+	}
+
+	// Get the new image URL
+	image := c.FormValue(dashboard_components.QuestionImageURL)
+	imageURL, _ := url.Parse(image)
+
+	// Set the image URL for the question
+	err = questions.SetImageByQuestionID(aah.sharedData.DB, &questionID, imageURL)
+	if err != nil {
```

Comment:
again maybe check for custom error if nothing was updated, otherwise return  err